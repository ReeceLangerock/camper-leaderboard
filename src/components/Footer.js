import React from "react";
import styled from "styled-components";

const FooterStyled = styled.div`
  background-color: #f4faff;
  color: #535657;
  font-size: 1.4rem;
  align-items: center;
  padding: 12px;
  align-content: center;
  text-align: center;
  vertical-align: middle;

  a {
    color: #3bb273;
    text-decoration: none;
  }
`;

const I = styled.i`
  margin-left: 10px;

  color: black;
  cursor: pointer;
`;

export class Footer extends React.Component {
  handleIconClick(link) {
    window.open("https://github.com/ReeceLangerock/camper-leaderboard", "_blank");
  }
  render() {
    const currentYear = new Date().getFullYear();

    return (
      <FooterStyled className="Footer">
        Â© {currentYear}{" "}
        <a href="http://reecelangerock.com" rel="noopener noreferrer" target="_blank">
          Reece Langerock
        </a>
        <I className={`fa fa-github`} aria-hidden="false" onClick={() => this.handleIconClick()} />
      </FooterStyled>
    );
  }
}
export default Footer;
