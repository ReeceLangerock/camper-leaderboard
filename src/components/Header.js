import React from 'react'

export class Header extends React.Component {
  render () {
    return (
      <div className='Header' ><h1>FCC Camper Leaderboard</h1></div>
    )
  }
}
export default Header
