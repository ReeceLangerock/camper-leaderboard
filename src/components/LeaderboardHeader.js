import React from 'react'
import { connect } from 'react-redux'
import * as actions from '../actions/actions'

export class LeaderboardHeader extends React.Component {
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick (e) {
    console.log()
    this.props.setSelection(e.target.value)
    this.props.getRecentCamperData()
  }

  render () {
    return <div className='LeaderboardHeader'>
      <button value='recent' onClick={this.handleClick}>Recent</button>
      <button value='alltime' onClick={this.handleClick}>All-Time</button>
    </div>
  }
}

const mapStateToProps = () => ({

})
export default connect(mapStateToProps, actions)(LeaderboardHeader)
