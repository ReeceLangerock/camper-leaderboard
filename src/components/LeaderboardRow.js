import React from "react";
import styled from "styled-components";

const Image = styled.img`
  max-height: 40px;
  width: 40px;
  margin-right: 5px;
  border-radius: 3px;
  border: 1px solid green;
  box-sizing: border-box;
`;

const Col = styled.td`
  padding: 5px;
  border-right: 1px solid;
`;
const Camper = Col.extend`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  text-align: left;

`;

const Row = styled.tr`
  &:nth-child(even) {
    background-color: #DEE7E7;
  }
`;

export class LeaderboardRow extends React.Component {
  render() {
    const camper = this.props.camper;
    return (
      <Row>
        <Col key ={'rank'} >{this.props.rank}</Col>
        <Camper key ={'camper'}>
          <Image src={camper.img} /> {camper.username}
        </Camper>
        <Col key ={'alltime'}>{camper.alltime}</Col>
        <Col key ={'recent'}>{camper.recent}</Col>
      </Row>
    );
  }
}
export default LeaderboardRow;
