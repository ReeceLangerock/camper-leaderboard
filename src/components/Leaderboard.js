import React from 'react'
import LeaderboardHeader from './LeaderboardHeader'
import LeaderboardRow from './LeaderboardRow'
import { connect } from 'react-redux'

import * as actions from '../actions/actions'

export class Leaderboard extends React.Component {

  componentWillMount () {}

  renderLeaderboardRow () {
    return <LeaderboardRow />
  }
  render () {
    return (
      <div className='Leaderboard'>
        <h1>{this.props.selection}</h1>
        <LeaderboardHeader />
        {this.renderLeaderboardRows}
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  selection: state.camperReducer.selection,
  recentCampers: state.camperReducer.recentCampers,
  allTimeCampers: state.camperReducer.allTimeCampers
})
export default connect(mapStateToProps, actions)(Leaderboard)
