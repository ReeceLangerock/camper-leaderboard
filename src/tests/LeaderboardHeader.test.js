import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, find, render, simulate } from "enzyme";
import enzymeSerializer from "enzyme-to-json/serializer";

import {LeaderboardHeader} from "./../components/LeaderboardHeader";
import configureStore from "redux-mock-store";

const initialState = {};
let mockStore = configureStore(initialState);
expect.addSnapshotSerializer(enzymeSerializer);
describe("LeaderboardHeader", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<LeaderboardHeader store={mockStore} />, div);
  });

  let component;
  beforeEach(() => {
    const props = {};
    component = mount(<LeaderboardHeader store={mockStore} {...props} />);
  });

  it("has the correct class", () => {
    expect(component.hasClass("LeaderboardHeader")).toEqual(true);
  });

  describe("Snapshot LeaderboardHeader", () => {
    it("matches snapshot", () => {
      const comp = mount(<LeaderboardHeader store={mockStore} />);
      expect(comp).toMatchSnapshot();
    });
  });
});
