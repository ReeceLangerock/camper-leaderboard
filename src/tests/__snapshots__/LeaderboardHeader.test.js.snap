// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardHeader Snapshot LeaderboardHeader matches snapshot 1`] = `
<LeaderboardHeader
  store={[Function]}
>
  <div
    className="LeaderboardHeader"
  >
    <button
      onClick={[Function]}
      value="recent"
    >
      Recent
    </button>
    <button
      onClick={[Function]}
      value="alltime"
    >
      All-Time
    </button>
  </div>
</LeaderboardHeader>
`;
